src/router_core/connections.c

```diff
@@ -473,7 +474,8 @@ void qdr_link_second_attach(qdr_link_t *link, qdr_terminus_t *source, qdr_termin
     action->args.connection.link   = link;
     action->args.connection.source = source;
     action->args.connection.target = target;
-    qdr_action_enqueue(link->core, action);
+    if (link)
```

Comment:
I wonder if the test might be better done in AMQP_link_attach_handler in router_node.c? That is the only place that calls the method, and is the one that retrieves the null context.

Reply:
I added the null check in AMQP_link_attach_handler and in qdr_link_second_attach() just to be doubly sure.